torch/fx/experimental/symbolic_shapes.py

```diff
@@ -271,6 +271,11 @@ def _nyi():
 
 float_magic_methods = {"add", "sub", "mul", "truediv", "ceil", "floor", "eq", "gt", "lt", "le", "ge", "pow"}
 
+magic_methods_on_builtins = {"min", "max"}
+magic_methods_on_math = {"min", "max"}
+always_float_magic_methods = {"truediv", "sym_float"}
```

Comment:
Current truediv leads to
```python
======================================================================
ERROR: test_method_fn_floordiv_first_type_int_second_type_float (__main__.TestSymNumberMagicMethods)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/albandes/local/pytorch/3.8_debug_source/torch/testing/_internal/common_utils.py", line 248, in instantiated_test
    test(self, **param_kwargs)
  File "test_dynamic_shapes.py", line 538, in test_method
    self._do_test(fn, inp1, inp2, shape_env, is_unary_fn)
  File "test_dynamic_shapes.py", line 498, in _do_test
    self.assertEqual(guard_fn(out), ref_out)
  File "test_dynamic_shapes.py", line 479, in guard_fn
    raise e
  File "test_dynamic_shapes.py", line 474, in guard_fn
    return getattr(v.node, f"guard_{tp}")("", 0)
  File "/home/albandes/local/pytorch/3.8_debug_source/torch/fx/experimental/symbolic_shapes.py", line 188, in guard_int
    return int(self.shape_env.evaluate_expr(self.expr))
  File "/home/albandes/local/pytorch/3.8_debug_source_env/lib/python3.8/site-packages/sympy/core/expr.py", line 321, in __int__
    r = self.round(2)
  File "/home/albandes/local/pytorch/3.8_debug_source_env/lib/python3.8/site-packages/sympy/core/expr.py", line 3817, in round
    raise TypeError(
TypeError: Expected a number but got FloorDiv:
```

Also current impl doesn't seem to follow best practices as it does fusion which is explicitly mentioned as a bad idea: https://docs.sympy.org/latest/guides/custom-functions.html#best-practices-for-eval